node_group_advanced.tf

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright 2020 Quortex
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+# Common resources
+
+resource "aws_iam_instance_profile" "quortex" {
+  name = var.instance_profile_name
+  role = aws_iam_role.quortex_role_worker.name
+}
+
+
+# One launch template per node group
+resource "aws_launch_template" "quortex_launch_tpl" {
+  for_each = var.node_groups_advanced
+
+  name = lookup(each.value, "asg_name", "${var.cluster_name}_${each.key}")
+
+  image_id = each.value.image_id
+
+  user_data = base64encode(
+    templatefile(
+      "${path.module}/userdata.sh.tpl",
+      {
+        cluster_name            = aws_eks_cluster.quortex.name
+        base64_cluster_ca       = aws_eks_cluster.quortex.certificate_authority[0].data
+        api_server_url          = aws_eks_cluster.quortex.endpoint
+        kubelet_more_extra_args = ""
+        node_taints             = length(each.value.taints) == 0 ? "" : join(",", [for k, v in lookup(each.value, "taints", {}) : "${k}=${v}"])
+        node_labels = join(
+          ",",
+          [
+            for k, v in
+            merge(
+              # Built-in labels
+              map(
+                "eks.amazonaws.com/nodegroup-image", each.value.image_id,
+                "eks.amazonaws.com/nodegroup", each.key,
+                "nodegroup", each.key
+              ),
+              # User-specified labels
+              lookup(each.value, "labels", {}),
+            )
+          : "${k}=${v}"]
+        )
+      }
+    )
+  )
+
+  block_device_mappings {
+    device_name = "/dev/xvda"
+
+    ebs {
```

Comment:
It would be better to be able to configure EBS attributes with variables

Reply:
Changed as suggested